src/agent/rustjail/src/mount.rs

```diff
@@ -288,6 +289,31 @@ pub fn init_rootfs(
         }
     }
 
+    if expose_configfs {
+        if let Some(annotations) = spec.annotations() {
+            annotations.iter().for_each(|(k, v)| {
+                if k == "io.katacontainers.pkg.oci.container_type" && v != "pod_sandbox" {
+                    if let Err(e) = mount(
+                        Some("configfs"),
+                        format!("{}/sys/kernel/config", rootfs).as_str(),
+                        Some("configfs"),
+                        MsFlags::MS_NODEV
+                            | MsFlags::MS_NOSUID
+                            | MsFlags::MS_NOEXEC
+                            | MsFlags::MS_RELATIME,
+                        None::<&str>,
+                    ) {
```

Comment:
@fidencio Any reason to mount `/sys/kernel/config` within `init_rootfs()` ? It seems this could be performed from the agent top level code directly. It would spare all the changes to propagate `agent.expose_configfs` down to here.

Reply:
It seems the most logical part to have this change, @gkurz.

Would you mind expand a little bit your suggestion so I could understand it better?

Reply:
I've privately spoken with @gkurz, and basically his suggestion is to try to address this somewhere else.

I think this can be done on the `containers.rs`, but not earlier than that, I will give it a try.